### ISSUE TITLE
False positive in the resource leak checker when the finalizer method has multiple @EnsuresCalledMethods annotation

### ISSUE THREAD

**Issue by Nargeshdb** [typetools/checker-framework#5911]
Consider the following example, in which there's a class that includes two fields with non-empty must-call obligations. The resource leak checker is unable to verify that both fields are properly released in the finalizer method.

```
import org.checkerframework.checker.mustcall.qual.*;
import org.checkerframework.checker.calledmethods.qual.*;

class MultipleECMAnnosOnMethods {

  @InheritableMustCall("a")
  static class Foo {
    void a() {}
  }

  @InheritableMustCall("b")
  static class Bar {
    void b() {}
  }

  @InheritableMustCall("finalizer")
  static class OwningField {

    // :: warning: (required.method.not.called)
    private final @Owning Foo owningFoo;

    // :: warning: (required.method.not.called)
    private final @Owning Bar owningBar;

    public OwningField() {
      this.owningFoo = new Foo();
      this.owningBar = new Bar();
    }

    @EnsuresCalledMethods(value = {"this.owningBar"}, methods = {"b"})
    @EnsuresCalledMethods(value = {"this.owningFoo"}, methods = {"a"})
    void finalizer() {
      this.owningFoo.a();
      this.owningBar.b();
    }

  }
}
```

**Comment by jacek-lewandowski** [comment]
Shouldn't it be written like:
```java
@EnsureCalledMethods.List({
    @EnsuresCalledMethods(value = {"this.owningBar"}, methods = {"b"})
    @EnsuresCalledMethods(value = {"this.owningFoo"}, methods = {"a"})
}
```
?

**Comment by msridhar** [comment]
> Shouldn't it be written like:
> 
> ```java
> @EnsureCalledMethods.List({
>     @EnsuresCalledMethods(value = {"this.owningBar"}, methods = {"b"})
>     @EnsuresCalledMethods(value = {"this.owningFoo"}, methods = {"a"})
> }
> ```
> 
> ?

@jacek-lewandowski since the `@EnsuresCalledMethod` annotation [has a `@Repeatable` meta-annotation](https://github.com/typetools/checker-framework/blob/df4246d1cb0c743c11b6319f8f18e347eaa84f0f/checker-qual/src/main/java/org/checkerframework/checker/calledmethods/qual/EnsuresCalledMethods.java#L43) I think either variant is equivalent.
